mindsdb/integrations/handlers/hive_handler/hive_handler.py

```diff
@@ -0,0 +1,191 @@
+)
+from mindsdb.integrations.libs.const import HANDLER_CONNECTION_ARG_TYPE as ARG_TYPE
+
+
+class HiveHandler(DatabaseHandler):
+    """
+    This handler handles connection and execution of the Hive SQL statements.
+    """
+
+    name = 'hive'
+
+    def __init__(self, name: str, connection_data: Optional[dict], **kwargs):
+        super().__init__(name)
+        self.parser = parse_sql
+        self.dialect = 'hive'
+
+        self.connection_data = connection_data
+        self.kwargs = kwargs
+
+        self.connection = None
+        self.is_connected = False
+
+    def __del__(self):
+        if self.is_connected is True:
+            self.disconnect()
+
+    def connect(self):
+        if self.is_connected is True:
+            return self.connection
+
+        config = {
+            'host': self.connection_data.get('host'),
+            'port': self.connection_data.get('port'),
+            'auth': self.connection_data.get('auth', 'CUSTOM'),
+            'username': self.connection_data.get('user'),
+            'password': self.connection_data.get('password'),
+            'database': self.connection_data.get('database')
+        }
+
+        connection = hive.Connection(**config)
+        self.is_connected = True
+        self.connection = connection
+        return self.connection
+
+    def disconnect(self):
+        if self.is_connected is False:
+            return
+        self.connection.close()
+        self.is_connected = False
+        return
+
+    def check_connection(self) -> StatusResponse:
+        """
+        Check the connection of the Hive database
+        :return: success status and error message if error occurs
+        """
+
+        response = StatusResponse(False)
+        need_to_close = self.is_connected is False
+
+        try:
+            self.connect()
+            response.success = True
+        except Exception as e:
+            log.error(f'Error connecting to Hive {self.connection_data["database"]}, {e}!')
+            response.error_message = str(e)
+
+        if response.success is True and need_to_close:
+            self.disconnect()
+        if response.success is False and self.is_connected is True:
+            self.is_connected = False
+
+        return response
+
+    def native_query(self, query: str) -> Response:
+        """
+        Receive SQL query and runs it
+        :param query: The SQL query to run in Hive
+        :return: returns the records from the current recordset
+        """
+
+        need_to_close = self.is_connected is False
+
+        connection = self.connect()
+        with connection.cursor() as cur:
+            try:
+                cur.execute(query)
+                result = cur.fetchall()
+                if result:
+                    response = Response(
+                        RESPONSE_TYPE.TABLE,
+                        pd.DataFrame(
+                            result,
+                            columns=[x[0].split('.')[1] for x in cur.description]
```

Comment:
If query='show tables', then cur.description[0] = 'tab_name' (without dot), therefore `.split('.')[1]` raise error. let change it to `.split('.')[-1]`, that should fix it.

Reply:
Hi @StpMax ,
Thank you for your observation. Find the fix in the latest commits.